plugins/modules/win_scoop.ps1

```diff
@@ -0,0 +1,234 @@
+  }
+
+  $res.stdout -split "`n" |
+  Select-String '(.*?) \(v:(.*?)\) \[(.*?)\]' |
+  ForEach-Object {
+    [PSCustomObject]@{
+      Package = $_.Matches[0].Groups[1].Value
+      Version = $_.Matches[0].Groups[2].Value
+      Bucket  = $_.Matches[0].Groups[3].Value
+    }
+  }
+}
+
+function Get-InstallScoopPackageArguments {
+  $arguments = [System.Collections.Generic.List[String]]@()
+
+  if ($architecture) {
+    $arguments.Add("--arch")
+    $arguments.Add($architecture)
+  }
+  if ($global) {
+    $arguments.Add("--global")
+  }
+  if ($independent) {
+    $arguments.Add("--independent")
+  }
+  if ($no_cache) {
+    $arguments.Add("--no-cache")
+  }
+  if ($skip_checksum) {
+    $arguments.Add("--skip")
+  }
+
+  return , $arguments
+}
+
+function Install-ScoopPackage {
+  param(
+    [Parameter(Mandatory = $true)] [string]$scoop_path,
+    [Parameter(Mandatory = $true)] [String[]]$packages
+  )
+  $arguments = [System.Collections.Generic.List[String]]@("powershell.exe", $scoop_path, "install")
+  $arguments.AddRange($packages)
+
+  $common_args = Get-InstallScoopPackageArguments
+  $arguments.AddRange($common_args)
+
+  $command = Argv-ToString -arguments $arguments
+
+  if (-not $module.CheckMode) {
+    $res = Run-Command -Command $command
+    $module.Result.rc = $res.rc
+
+    if ($module.Verbosity -gt 1) {
+      $module.Result.stdout = $res.stdout
+    }
+  }
+  $module.Result.changed = $true
+}
+
+function Get-UninstallScoopPackageArguments {
+  $arguments = [System.Collections.Generic.List[String]]@()
+
+  if ($independent) {
+    $arguments.Add("--global")
+  }
+  if ($purge) {
+    $arguments.Add("--purge")
+  }
+
+  return , $arguments
+}
+
+function Uninstall-ScoopPackage {
+  param(
+    [Parameter(Mandatory = $true)] [string]$scoop_path,
+    [Parameter(Mandatory = $true)] [String[]]$packages
+  )
+
+  $arguments = [System.Collections.Generic.List[String]]@("powershell.exe", $scoop_path, "uninstall")
+  $arguments.AddRange($packages)
+
+  $common_args = Get-UninstallScoopPackageArguments
+  $arguments.AddRange($common_args)
+
+  $command = Argv-ToString -arguments $arguments
+
+  if (-not $module.CheckMode) {
+    $res = Run-Command -Command $command
+    $module.Result.rc = $res.rc
+
+    if ($module.Verbosity -gt 1) {
+      $module.Result.stdout = $res.stdout
+    }
+    if (-not ($res.stdout -match "ERROR '(.*?)' isn't installed.")) {
+      $module.Result.changed = $true
+    }
+  }
+  else {
+    $module.Result.changed = $true
+  }
+}
+
+$scoop_path = Install-Scoop
+
+$installed_packages = Get-ScoopPackages -scoop_path $scoop_path
+
+if ($state -in @("absent")) {
+  # Always attempt uninstall
+  # Packages can be in a broken state where they don't appear scoop export
+  Uninstall-ScoopPackage -scoop_path $scoop_path -packages $name
+}
+
+if ($state -in @("present")) {
+  $missing_packages = foreach ($package in $name) {
+    if ($installed_packages.Package -notcontains $packages) {
```

Comment:
Looks like this is why the tests aren't passing, sorry for the faulty suggestion in the last round of commits.

```suggestion
    if ($installed_packages.Package -notcontains $package) {
```